src/Swarm/Game/Scenario/Cell.hs

```diff
@@ -20,15 +20,27 @@ import Swarm.Util.Yaml
 -- World cells
 ------------------------------------------------------------
 
--- | A single cell in a world map, which contains a terrain value,
---   and optionally an entity and robot.
-data Cell = Cell
+data ProtoCell e = Cell
   { cellTerrain :: TerrainType
-  , cellEntity :: Maybe Entity
+  , cellEntity :: Maybe e
   , cellRobots :: [IndexedTRobot]
   }
   deriving (Eq, Show)
 
+-- | A single cell in a world map, which contains a terrain value,
+--   and optionally an entity and robot.
+type Cell = ProtoCell Entity
+
+-- Note: This is used only for the purpose of WorldPalette
+instance ToJSON Cell where
```

Comment:
Since it appears naturally in the world palette it seems OK to me.

This is like having a warning on gardening tools that they are only supposed to be used safely in the garden.